src/Node/ForNode.php

```diff
@@ -25,8 +25,12 @@ class ForNode extends Node
 {
     protected $loop;
 
-    public function __construct(AssignNameExpression $keyTarget, AssignNameExpression $valueTarget, AbstractExpression $seq, AbstractExpression $ifexpr = null, \Twig_NodeInterface $body, \Twig_NodeInterface $else = null, $lineno, $tag = null)
+    public function __construct(AssignNameExpression $keyTarget, AssignNameExpression $valueTarget, AbstractExpression $seq, AbstractExpression $ifexpr = null, \Twig_NodeInterface $body = null, \Twig_NodeInterface $else = null, $lineno = 0, $tag = null)
     {
+        if (null === $body) {
+            throw new \LogicException('The "body" node of a "for" node cannot be null.');
```

Comment:
I would make this a `\TypeError`, so that you get the same exception type than what the right typehint would give.